app/kumactl/cmd/apply/apply.go

```diff
@@ -83,39 +83,41 @@ func NewApplyCmd(pctx *kumactl_cmd.RootContext) *cobra.Command {
 					}
 				}
 			}
+			resources := strings.Split(string(b), "---")
+			for _, resource := range resources {
+				bytes := template.Render(resource, ctx.args.vars)
+				res, err := rest.UnmarshallToCore(bytes)
+				if err != nil {
+					return errors.Wrap(err, "YAML contains invalid resource")
+				}
+				if err := mesh.ValidateMeta(res.GetMeta().GetName(), res.GetMeta().GetMesh(), res.Scope()); err.HasViolations() {
+					return err.OrNil()
+				}
 
-			bytes := template.Render(string(b), ctx.args.vars)
-			res, err := rest.UnmarshallToCore(bytes)
-			if err != nil {
-				return errors.Wrap(err, "YAML contains invalid resource")
-			}
-			if err := mesh.ValidateMeta(res.GetMeta().GetName(), res.GetMeta().GetMesh(), res.Scope()); err.HasViolations() {
-				return err.OrNil()
-			}
+				if ctx.args.dryRun {
+					p, err := printers.NewGenericPrinter(output.YAMLFormat)
+					if err != nil {
+						return err
+					}
+					if err := p.Print(rest_types.From.Resource(res), cmd.OutOrStdout()); err != nil {
+						return err
+					}
+					return nil
+				}
 
-			if ctx.args.dryRun {
-				p, err := printers.NewGenericPrinter(output.YAMLFormat)
+				var rs store.ResourceStore
```

Comment:
One change here please:
First prase all the resources in the loop, then do upsert using all of them, this way - when one resource is broken, we won't start sending requests to API Server.

Reply:
But, currently `resourceStore`'s `Create` method not accepts `resourceList`